src/plugins/intel_gna/src/transformations/matops_to_dwsc.cpp

```diff
@@ -0,0 +1,261 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+#include "matops_to_dwsc.hpp"
+
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/pass/graph_rewrite.hpp>
+#include <ngraph/pattern/op/or.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+#include <openvino/cc/ngraph/itt.hpp>
+
+namespace ov {
+namespace intel_gna {
+namespace pass {
+
+// Check if the previous node is convolution, groupconvolution or matmul followed by add
+static bool IsFusable(const Output<Node>& parent) {
+    if (nullptr == std::dynamic_pointer_cast<ngraph::opset1::GroupConvolution>(parent.get_node()->shared_from_this()) ||
+        nullptr == std::dynamic_pointer_cast<ngraph::opset1::Convolution>(parent.get_node()->shared_from_this()) ||
+        nullptr == std::dynamic_pointer_cast<ngraph::opset1::MatMul>(parent.get_node()->shared_from_this()))
+        return false;
+    return true;
+}
+
+static bool Decompose(std::shared_ptr<ov::Node> math_node) {
+    const Output<Node>& input = math_node->input_value(0);
+    const Output<Node>& params = math_node->input_value(1);
+
+    auto input_shape = input.get_shape();
+    auto params_shape = params.get_shape();
+    auto auto_broadcast = math_node->get_autob();
+    auto output_shape = math_node->get_output_shape(0);
+
+    if (input_shape.size() == 0)
+        return false;
+
+    uint64_t N, C, H, W;
```

Comment:
please assign at least `0` to avoid potential warnings or issues in the future.

Reply:
Done